techmatters/terraso-mobile-client
# fix: soil id data caching

## Description

App was locking up because multiple soil ID components were rendering at once, which wasn't a supported use-case in the initial loading implementation. This caused contention over the soil ID redux store, resulting in an infinite cycle of queries canceling each other out. Update the soil ID data hook to use the new cache-based implementation of the redux store slice.

### Checklist

### Related Issues
#1647 

### Verification steps

Open soil ID views through both the map and sites list. Verify that the app does not lock up anymore. Verify that site data is loaded once and continues to be available until the user navigates way from the site. Verify that data displayed on map callouts persists while the user views site data.
